test/integration/not-ready-test.js

```diff
@@ -0,0 +1,167 @@
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+'use strict';
+
+var test = require('tape');
+var _ = require('underscore');
+var async = require('async');
+
+var testRingpopCluster = require('../lib/test-ringpop-cluster.js');
+var sendPing = require('../../lib/gossip/ping-sender.js');
+var sendPingReqs = require('../../lib/gossip/ping-req-sender.js');
+var Member = require('../../lib/membership/member.js');
+
+testRingpopCluster({
+    size: 4,
+    boostrapSize: 3,
+    tap: function(cluster) {
+        cluster[3].bootstrapHosts = ['127.0.0.1:12345', cluster[3].whoami()]
+    },
+    waitForConvergence: false
+}, 'not-ready error on ping', function t(bootRes, cluster, assert) {
+    assert.plan(1);
+
+    var ringpop = cluster[3];
+    cluster.forEach(function eachNode(node) {
+        node.membership.makeAlive(ringpop.whoami(), ringpop.incarnationNumber);
+    });
+    var unreachableMember = cluster[0].membership.findMemberByAddress(ringpop.whoami());
+
+    sendPing({
+        ringpop: cluster[0],
+        target: unreachableMember
+    }, function onPing(err, res) {
+        assert.equal(err.type, 'ringpop.not-ready');
+        assert.end();
+    });
+});
+
+testRingpopCluster({
+    size: 4,
+    boostrapSize: 3,
+    tap: function(cluster) {
+        cluster[3].bootstrapHosts = ['127.0.0.1:12345', cluster[3].whoami()]
+    },
+    waitForConvergence: false
+}, 'ping-req fails when not ready', function t(bootRes, cluster, assert) {
+    assert.plan(2);
+
+    var ringpop = cluster[3];
+    cluster.forEach(function eachNode(node) {
+        node.membership.makeAlive(ringpop.whoami(), ringpop.incarnationNumber);
+    });
+    var unreachableMember = cluster[0].membership.findMemberByAddress(ringpop.whoami());
+
+    sendPingReqs({
+        ringpop: cluster[0],
+        unreachableMember: unreachableMember,
+        pingReqSize: 2
+    }, function onPing(err, res) {
+        res.pingReqErrs.forEach(function each(pingReqRes) {
+            assert.equals(pingReqRes.pingStatus, false, 'ping status of node that is not ready is false');
+        });
+        assert.end();
+    });
+});
+
+testRingpopCluster({
+    size: 4,
+    boostrapSize: 2,
+    tap: function(cluster) {
+        cluster[2].bootstrapHosts = ['127.0.0.1:12345', cluster[2].whoami()]
+        cluster[3].bootstrapHosts = ['127.0.0.1:12345', cluster[3].whoami()]
+    },
+    waitForConvergence: false
+}, 'ping-req inconclusive when all ping-req nodes are not ready', function t(bootRes, cluster, assert) {
+    assert.plan(1);
+
+    var ringpop = cluster[3];
```

Comment:
This test is hard to read/understand. Consider clarifying, naming better, refactoring, e.g.:

```
// This Ringpop is partially crippled. It will be added to the cluster's membership, but will be unable to join the cluster. Below, it'll be selected for ping-req, but expected to respond with a not-ready error.
var reachableRingpop = cluster[3];
addMemberToCluster(cluster, reachableRingpop);

var pingReqRingpop = cluster[0];
var unreachableRingpop = cluster[1];
var unreachableMember = pingReqRingpop.membership.findMemberByAddress(unreachableRingpop.whoami());

// etc...
```